pkg/metrics/prometheus.go

```diff
@@ -0,0 +1,152 @@
+package metrics
+
+import (
+	"github.com/grafana/grafana/pkg/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"net/http"
+	"strings"
+)
+
+const (
+	namespace = "grafana"
+	subsystem = "backend"
+)
+
+type PrometheusPublisher struct {
+	CounterCollectors map[string]*prometheus.CounterVec
+	GaugeCollectors   map[string]*prometheus.GaugeVec
+}
+
+func CreatePrometheusPublisher() (*PrometheusPublisher, error) {
+	go func() {
+		http.Handle("/metrics", prometheus.Handler())
+		err := http.ListenAndServe(":9200", nil)
```

Comment:
> So if you query with a large step, values in between steps will be ignored. Is this not true for rate function due to it needing to detect rests?

If the step is larger than the rate period, you'll ignore some data.

> For example if you query increase(logins.count[1d]) over last 2 months with step of 60 (or a 100). will that be accurate? won't the step parameter need to be as low as sample rate to detect resets and calculate rate accurately?

Step is orthogonal to query evaluation. What you'll do is evaluate `increase(logins.count[1d])` independently at each of 2*30*24*60=86400 points in time. Each of those results would be accurate for its 1 day window.

(In reality such a query would be rejected due to a safety feature to prevent someone accidentally requesting too many steps).